commit 3327989a51f7d46d854c89bb6047dd68f843a7fd
Author: Brett Rudley <brudley@broadcom.com>
Date:   Fri Oct 1 18:03:27 2010 -0700

    staging: brcm80211: Move #includes out of header
    
    Start the process of moving #includes out of headers and into individual C files.
    For now, this patch addresses the softmac side of the driver, fullmac still to
    be done.
    
    Signed-off-by: Brett Rudley <brudley@broadcom.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

--- a/drivers/staging/brcm80211/include/linux_osl.h
+++ b/drivers/staging/brcm80211/include/linux_osl.h
@@ -17,7 +17,9 @@
 #ifndef _linux_osl_h_
 #define _linux_osl_h_
 
+#ifdef BRCM_FULLMAC
 #include <typedefs.h>
+#endif
 
 /* Linux Kernel: File Operations: start */
 extern void *osl_os_open_image(char *filename);
@@ -150,7 +152,9 @@ extern void osl_dma_unmap(osl_t *osh, ui
 
 /* register access macros */
 #if defined(BCMSDIO)
+#ifdef BRCM_FULLMAC
 #include <bcmsdh.h>
+#endif
 #define OSL_WRITE_REG(osh, r, v) (bcmsdh_reg_write(NULL, (uintptr)(r), sizeof(*(r)), (v)))
 #define OSL_READ_REG(osh, r) (bcmsdh_reg_read(NULL, (uintptr)(r), sizeof(*(r))))
 #endif
@@ -173,8 +177,10 @@ extern int osl_error(int bcmerror);
 
 #define OSL_SYSUPTIME()		((uint32)jiffies * (1000 / HZ))
 #define	printf(fmt, args...)	printk(fmt , ## args)
+#ifdef BRCM_FULLMAC
 #include <linux/kernel.h>	/* for vsn/printf's */
 #include <linux/string.h>	/* for mem*, str* */
+#endif
 /* bcopy's: Linux kernel doesn't provide these (anymore) */
 #define	bcopy(src, dst, len)	memcpy((dst), (src), (len))
 #define	bcmp(b1, b2, len)	memcmp((b1), (b2), (len))
@@ -320,7 +326,9 @@ extern int osl_error(int bcmerror);
 #define	W_SM(r, v)		(*(r) = (v))
 #define	BZERO_SM(r, len)	memset((r), '\0', (len))
 
+#ifdef BRCM_FULLMAC
 #include <linuxver.h>		/* use current 2.4.x calling conventions */
+#endif
 
 /* packet primitives */
 #define	PKTGET(osh, len, send)		osl_pktget((osh), (len))
--- a/drivers/staging/brcm80211/phy/wlc_phy_cmn.c
+++ b/drivers/staging/brcm80211/phy/wlc_phy_cmn.c
@@ -16,8 +16,11 @@
 
 #include <wlc_cfg.h>
 
+#include <typedefs.h>
 #include <osl.h>
 #include <linux/kernel.h>
+#include <linux/string.h>
+#include <linuxver.h>
 #include <bcmendian.h>
 #include <bcmnvram.h>
 #include <sbchipc.h>
--- a/drivers/staging/brcm80211/phy/wlc_phy_lcn.c
+++ b/drivers/staging/brcm80211/phy/wlc_phy_lcn.c
@@ -17,6 +17,9 @@
 #include <wlc_cfg.h>
 #include <qmath.h>
 #include <osl.h>
+#include <linux/kernel.h>
+#include <linux/string.h>
+#include <linuxver.h>
 #include <siutils.h>
 #include <bitfuncs.h>
 #include <hndpmu.h>
--- a/drivers/staging/brcm80211/phy/wlc_phy_n.c
+++ b/drivers/staging/brcm80211/phy/wlc_phy_n.c
@@ -15,6 +15,10 @@
  */
 
 #include <wlc_cfg.h>
+#include <typedefs.h>
+#include <linux/kernel.h>
+#include <linux/string.h>
+#include <linuxver.h>
 #include <osl.h>
 #include <siutils.h>
 #include <sbchipc.h>
--- a/drivers/staging/brcm80211/sys/wlc_alloc.c
+++ b/drivers/staging/brcm80211/sys/wlc_alloc.c
@@ -17,6 +17,9 @@
 #include <wlc_cfg.h>
 #include <typedefs.h>
 #include <bcmdefs.h>
+#include <linux/kernel.h>
+#include <linux/string.h>
+#include <linuxver.h>
 #include <osl.h>
 #include <bcmutils.h>
 #include <siutils.h>
--- a/drivers/staging/brcm80211/sys/wlc_ampdu.c
+++ b/drivers/staging/brcm80211/sys/wlc_ampdu.c
@@ -16,6 +16,7 @@
 
 #include <wlc_cfg.h>
 #include <typedefs.h>
+#include <linuxver.h>
 #include <bcmdefs.h>
 #include <osl.h>
 #include <bcmutils.h>
--- a/drivers/staging/brcm80211/sys/wlc_antsel.c
+++ b/drivers/staging/brcm80211/sys/wlc_antsel.c
@@ -20,6 +20,8 @@
 
 #include <typedefs.h>
 #include <qmath.h>
+#include <linux/kernel.h>
+#include <linuxver.h>
 #include <bcmdefs.h>
 #include <osl.h>
 #include <bcmutils.h>
--- a/drivers/staging/brcm80211/sys/wlc_bmac.c
+++ b/drivers/staging/brcm80211/sys/wlc_bmac.c
@@ -20,6 +20,7 @@
 
 #include <wlc_cfg.h>
 #include <typedefs.h>
+#include <linuxver.h>
 #include <bcmdefs.h>
 #include <osl.h>
 #include <linux/kernel.h>
--- a/drivers/staging/brcm80211/sys/wlc_channel.c
+++ b/drivers/staging/brcm80211/sys/wlc_channel.c
@@ -18,6 +18,9 @@
 #include <typedefs.h>
 #include <bcmdefs.h>
 #include <osl.h>
+#include <linux/kernel.h>
+#include <linux/string.h>
+#include <linuxver.h>
 #include <bcmutils.h>
 #include <siutils.h>
 #include <wlioctl.h>
--- a/drivers/staging/brcm80211/sys/wlc_event.c
+++ b/drivers/staging/brcm80211/sys/wlc_event.c
@@ -15,6 +15,8 @@
  */
 
 #include <typedefs.h>
+#include <linux/kernel.h>
+#include <linuxver.h>
 #include <bcmutils.h>
 #include <siutils.h>
 #include <bcmendian.h>
--- a/drivers/staging/brcm80211/sys/wlc_mac80211.c
+++ b/drivers/staging/brcm80211/sys/wlc_mac80211.c
@@ -16,6 +16,8 @@
 
 #include <wlc_cfg.h>
 #include <typedefs.h>
+#include <linux/kernel.h>
+#include <linuxver.h>
 #include <bcmdefs.h>
 #include <osl.h>
 #include <linux/ctype.h>
--- a/drivers/staging/brcm80211/sys/wlc_phy_shim.c
+++ b/drivers/staging/brcm80211/sys/wlc_phy_shim.c
@@ -23,6 +23,8 @@
 
 #include <wlc_cfg.h>
 #include <typedefs.h>
+#include <linuxver.h>
+#include <linux/kernel.h>
 #include <bcmutils.h>
 #include <bcmdefs.h>
 #include <osl.h>
--- a/drivers/staging/brcm80211/sys/wlc_rate.c
+++ b/drivers/staging/brcm80211/sys/wlc_rate.c
@@ -17,6 +17,8 @@
 #include <wlc_cfg.h>
 #include <typedefs.h>
 #include <osl.h>
+#include <linux/kernel.h>
+#include <linuxver.h>
 #include <bcmutils.h>
 #include <siutils.h>
 #include <bcmendian.h>
--- a/drivers/staging/brcm80211/sys/wlc_stf.c
+++ b/drivers/staging/brcm80211/sys/wlc_stf.c
@@ -16,6 +16,8 @@
 
 #include <wlc_cfg.h>
 #include <typedefs.h>
+#include <linux/kernel.h>
+#include <linuxver.h>
 #include <bcmdefs.h>
 #include <osl.h>
 #include <bcmutils.h>
--- a/drivers/staging/brcm80211/util/aiutils.c
+++ b/drivers/staging/brcm80211/util/aiutils.c
@@ -17,6 +17,9 @@
 #include <typedefs.h>
 #include <bcmdefs.h>
 #include <osl.h>
+#include <linux/kernel.h>
+#include <linux/string.h>
+#include <linuxver.h>
 #include <bcmutils.h>
 #include <siutils.h>
 #include <hndsoc.h>
--- a/drivers/staging/brcm80211/util/bcmotp.c
+++ b/drivers/staging/brcm80211/util/bcmotp.c
@@ -17,6 +17,9 @@
 #include <typedefs.h>
 #include <bcmdefs.h>
 #include <osl.h>
+#include <linux/kernel.h>
+#include <linux/string.h>
+#include <linuxver.h>
 #include <bcmdevs.h>
 #include <bcmutils.h>
 #include <siutils.h>
--- a/drivers/staging/brcm80211/util/bcmsrom.c
+++ b/drivers/staging/brcm80211/util/bcmsrom.c
@@ -18,6 +18,8 @@
 #include <bcmdefs.h>
 #include <osl.h>
 #include <linux/kernel.h>
+#include <linux/string.h>
+#include <linuxver.h>
 #include <stdarg.h>
 #include <bcmutils.h>
 #include <hndsoc.h>
--- a/drivers/staging/brcm80211/util/bcmutils.c
+++ b/drivers/staging/brcm80211/util/bcmutils.c
@@ -20,6 +20,8 @@
 #include <osl.h>
 #include <linux/ctype.h>
 #include <linux/kernel.h>
+#include <linux/string.h>
+#include <linuxver.h>
 #include <bcmutils.h>
 #include <siutils.h>
 #include <bcmnvram.h>
--- a/drivers/staging/brcm80211/util/bcmwpa.c
+++ b/drivers/staging/brcm80211/util/bcmwpa.c
@@ -14,6 +14,10 @@
  * CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
  */
 
+#include <typedefs.h>
+#include <linux/kernel.h>
+#include <linux/string.h>
+#include <linuxver.h>
 #include <bcmutils.h>
 #include <bcmwpa.h>
 
--- a/drivers/staging/brcm80211/util/hnddma.c
+++ b/drivers/staging/brcm80211/util/hnddma.c
@@ -15,6 +15,9 @@
  */
 
 #include <typedefs.h>
+#include <linux/kernel.h>
+#include <linux/string.h>
+#include <linuxver.h>
 #include <bcmdefs.h>
 #include <bcmdevs.h>
 #include <osl.h>
--- a/drivers/staging/brcm80211/util/hndpmu.c
+++ b/drivers/staging/brcm80211/util/hndpmu.c
@@ -17,6 +17,8 @@
 #include <typedefs.h>
 #include <bcmdefs.h>
 #include <osl.h>
+#include <linux/string.h>
+#include <linuxver.h>
 #include <linux/kernel.h>
 #include <bcmutils.h>
 #include <siutils.h>
--- a/drivers/staging/brcm80211/util/nicpci.c
+++ b/drivers/staging/brcm80211/util/nicpci.c
@@ -15,6 +15,8 @@
  */
 
 #include <typedefs.h>
+#include <linux/string.h>
+#include <linuxver.h>
 #include <bcmdefs.h>
 #include <osl.h>
 #include <bcmutils.h>
--- a/drivers/staging/brcm80211/util/nvram/nvram_ro.c
+++ b/drivers/staging/brcm80211/util/nvram/nvram_ro.c
@@ -16,6 +16,7 @@
 
 #include <typedefs.h>
 #include <bcmdefs.h>
+#include <linux/string.h>
 #include <osl.h>
 #include <bcmutils.h>
 #include <siutils.h>
--- a/drivers/staging/brcm80211/util/siutils.c
+++ b/drivers/staging/brcm80211/util/siutils.c
@@ -17,6 +17,9 @@
 #include <typedefs.h>
 #include <bcmdefs.h>
 #include <osl.h>
+#include <linux/kernel.h>
+#include <linux/string.h>
+#include <linuxver.h>
 #include <bcmutils.h>
 #include <siutils.h>
 #include <bcmdevs.h>
